src/main.c

```diff
@@ -1,4 +1,26 @@
-int main () {
-	while(1);
+#include <stdio.h>
+#include "app/app_header.h"
+
+bool check_app_version(const app_header_t header) {
+	if (header.magic != APP_HEADER_MAGIC) {
+		return false;
+	}
+	if (header.version.major != APP_HEADER_VERSION_MAJOR) {
+		return false;
+	}
+	if (header.version.minor != APP_HEADER_VERSION_MINOR) {
+		return false;
+	}
+	return true;
```

Comment:
```suggestion
	}
	
	if (header.version.major != APP_HEADER_VERSION_MAJOR) {
		return false;
	}

	if (header.version.minor < APP_HEADER_VERSION_MINOR) {
		return false;
	}
	
	if (header.version.patch < APP_HEADER_VERSION_PATCH) {
		return false;
	}

	return true;
```
Remember semver, only a major change will break forwards compatibility on the side of the bootloader because it would mean the fields of the header no longer reside where they originally did. A minor update would imply a new field is added, but didn't break backward compatibility. A patch would be something small like renaming a field but retaining its functionality in full (i.e `entry_point` -> `app_start`).